File: examples/ejemplo.py
# ─────────────────────────────────────────────────────────────
# ejemplo.py — Fichero de prueba para el analizador de Python.
#
# Contiene funciones con distintas complejidades para
# verificar que el analizador las detecta correctamente:
#   - constante_ejemplo()    → O(1)
#   - busqueda_lineal()      → O(n)
#   - burbuja()              → O(n²)
#   - factorial_recursivo()  → O(n) recursivo
# ─────────────────────────────────────────────────────────────


def constante_ejemplo(x: int) -> int:
    """Operación constante: no depende del tamaño de la entrada."""
    return x * 2 + 1


def busqueda_lineal(lista: list[int], objetivo: int) -> int:
    """Busca un elemento recorriendo la lista una vez → O(n)."""
    for i in range(len(lista)):
        if lista[i] == objetivo:
            return i
    return -1


def burbuja(lista: list[int]) -> list[int]:
    """Ordenamiento burbuja clásico → O(n²)."""
    n = len(lista)
    for i in range(n):
        for j in range(0, n - i - 1):
            if lista[j] > lista[j + 1]:
                lista[j], lista[j + 1] = lista[j + 1], lista[j]
    return lista


def factorial_recursivo(n: int) -> int:
    """Calcula el factorial recursivamente → O(n) recursivo."""
    if n <= 1:
        return 1
    return n * factorial_recursivo(n - 1)


if __name__ == "__main__":
    datos = [64, 34, 25, 12, 22, 11, 90]
    print("Original:", datos)
    print("Ordenado:", burbuja(datos.copy()))
    print("Factorial de 5:", factorial_recursivo(5))

File: examples/ejemplo.java
/*
 * ejemplo.java — Fichero de prueba para el analizador de Java.
 *
 * Contiene métodos con distintas complejidades:
 *   - sumaConstante()       → O(1)
 *   - busquedaLineal()      → O(n)
 *   - ordenarBurbuja()      → O(n²)
 *   - factorialRecursivo()  → O(n) recursivo
 */

public class ejemplo {

    /* Operación constante: no depende del tamaño de entrada */
    public static int sumaConstante(int a, int b) {
        return a + b;
    }

    /* Busca un elemento recorriendo el array una vez → O(n) */
    public static int busquedaLineal(int[] arr, int objetivo) {
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] == objetivo) {
                return i;
            }
        }
        return -1;
    }

    /* Ordenamiento burbuja clásico → O(n²) */
    public static void ordenarBurbuja(int[] arr) {
        int n = arr.length;
        for (int i = 0; i < n - 1; i++) {
            for (int j = 0; j < n - i - 1; j++) {
                if (arr[j] > arr[j + 1]) {
                    int temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                }
            }
        }
    }

    /* Factorial recursivo → O(n) recursivo */
    public static int factorialRecursivo(int n) {
        if (n <= 1) {
            return 1;
        }
        return n * factorialRecursivo(n - 1);
    }

    public static void main(String[] args) {
        int[] datos = {64, 34, 25, 12, 22, 11, 90};

        System.out.println("Factorial de 5: " + factorialRecursivo(5));

        ordenarBurbuja(datos);
        System.out.print("Array ordenado: ");
        for (int dato : datos) {
            System.out.print(dato + " ");
        }
        System.out.println();
    }
}
